addons/account/wizard/account_move_send.py

```diff
@@ -317,7 +322,30 @@ def _prepare_invoice_pdf_report(self, invoice, invoice_data):
         if invoice.invoice_pdf_report_id:
             return
 
-        content, _report_format = self.env['ir.actions.report']._render('account.account_invoices', invoice.ids)
+        # Render the invoice PDF but allow to set a custom report_name and custom values.
+        # That way, all invoice reports are separated from the main one but don't need an
+        # extra report every time.
+        IrActionsReport = type(self.env['ir.actions.report'])
+        _render_template = IrActionsReport._render_template
+        inv_report, inv_report_values = self._get_invoice_pdf_report_to_render(invoice, invoice_data)
+
+        def render_template(records, template, values=None):
+            return _render_template(
+                records,
+                template,
+                values={
+                    **(values or {}),
+                    **inv_report_values,
+                },
+            )
+
+        with patch.object(IrActionsReport, '_render_template', side_effect=render_template, autospec=True):
```

Comment:
Goodness. I really hope this is some kind of temporary band-aid while we wait for another task to correctly redo the invoice report generation...

Reply:
@antoine162 Yeah... I know... I don't really want to refactor the whole PDF generation just because I don't want to define a new ir.actions.report everytime I do a custom PDF report I don't want visible on the view :/

Reply:
:see_no_evil: